saleor/payment/gateways/stripe/webhooks.py

```diff
@@ -158,6 +130,15 @@ def _finalize_checkout(
     )
 
 
+def _get_checkout(payment_id: int) -> Optional[Checkout]:
+    return (
+        Checkout.objects.prefetch_related("payments")
+        .select_for_update(of=("self",))
+        .filter(payments__id=payment_id, payments__is_active=True)
+        .first()
+    )
+
+
 def _update_payment_with_new_transaction(
```

Comment:
I believe a name should be changed. Previously we always upade a payment with new transaction. Currently only when condition inside a function is true